lighthouse-core/test/audits/third-party-summary-test.js

```diff
@@ -81,4 +84,42 @@ describe('Third party summary', () => {
       notApplicable: true,
     });
   });
+
+  it('does not return third party entity that matches main resource entity', async () => {
+    const artifacts = {
+      devtoolsLogs: {
+        defaultPass: networkRecordsToDevtoolsLog([
+          {url: 'https://facebook.com'},
+          {url: 'http://photos-c.ak.fbcdn.net/photos-ak-sf2p/photo.jpg'},
+        ]),
+      },
+      traces: {defaultPass: pwaTrace},
+      URL: {finalUrl: 'https://facebook.com'},
+    };
+
+    const results = await ThirdPartySummary.audit(artifacts, {computedCache: new Map()});
+
+    expect(results.details.items).toEqual([
```

Comment:
to make the test super obvious and narrow to why we have it, how about something more like...

```js
const externalArtifacts = {...artifacts, URL: {finalUrl: 'http://example.com'}};
const resultsOnExternal = await ThirdPartySummary.audit(externalArtifacts);
const facebookArtifacts = {...artifacts, URL: {finalUrl: 'http://facebook.com'}};
const resultsOnFacebook = await ThirdPartySummary.audit(facebookArtifacts);

const externalEntities = resultsOnExternal.details.items.map(item => item.entity.text);
const facebookEntities = resultsOnFacebook.details.items.map(item => item.entity.text);
expect(externalEntities).toEqual(['Facebook', 'Google Tag Manager', 'Google Analytics'])
expect(facebookEntities).toEqual(['Google Tag Manager', 'Google Analytics'])
```

right now it's kinda hard to trust that the records wouldn't just be ignored to begin with because we don't set any explicit transfer size or anything :)